### PR TITLE
Dependencies.kt

### PR DESCRIPTION
**Pull request by floschu**
All Project- and Module-Dependencies are now located in a `Dependencies.kt` file located in `buildSrc`.

#### Why?
[Click me](https://handstandsam.com/2018/02/11/kotlin-buildsrc-for-better-gradle-dependency-management/)

### CODE REVIEW

**Review by FlowMo7 — COMMENTED** [review]
I generally like it, however I'm not that big of a fan of the maps within the Dependencies.kt file yet.
I get that they are the easiest way to get the dependencies named somewhere with the possibility to just call `.values()` and get all dependencies, however, they destroy the whole autocomplete flow as the key in the map cannot be autocompleted, so the key names would actually still have to be looked up in the dependencies file.

As a possible solution to that, maybe we could make all dependencies as `const val`s in their respective objects, and provide a `val allDependencies: List<String>` getter, which substitutes the current `.values()` behaviour.

(Or the approach to have/use a `.values()` method in the first place could be challanged, so that each dependency would needed to be included explicitely)

**Comment by floschu** [comment]
Yes it would probably also be less convoluted without the `.values()` 👍

**Comment by mariohahn** [comment]
#würde aber darf nicht

File: buildSrc/src/main/kotlin/Dependencies.kt [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+
+    const val Gson = "com.google.code.gson:gson:2.8.5"
+    const val Timber = "com.jakewharton.timber:timber:4.7.1"
+    const val Threeten = "com.jakewharton.threetenabp:threetenabp:1.2.1"
+
+    const val LeakCanaryOp = "com.squareup.leakcanary:leakcanary-android:${Versions.Leakcanary}"
+    const val LeakCanaryNoOp = "com.squareup.leakcanary:leakcanary-android-no-op:${Versions.Leakcanary}"
+}
+
+enum class AndroidAppUtil(val dep: String): DependencyItem by dep {
```
Comment (FlowMo7) [inline buildSrc/src/main/kotlin/Dependencies.kt:49]: I would just like to bring up the discussion whether we want to keep the val name short as `dep`, or if we could spare a `dependency` here.. :thinking:

(I'm not sure which one I would consider better, but as `dep` is (at least for me) not that familiar as an abbreviation to `dependency`, I'm slightly tending to `dependency`...)

```kotlin
enum class AndroidAppUtil(val dependency: String): DependencyItem by dependency {
    UI("com.tailoredapps.androidutil:util-ui:${Versions.AndroidAppUtil}");
    
    override fun toString() = dependency
}
```
vs.
```kotlin
enum class AndroidAppUtil(val dep: String): DependencyItem by dep {
    UI("com.tailoredapps.androidutil:util-ui:${Versions.AndroidAppUtil}");
    
    override fun toString() = dep
}
```

File: buildSrc/src/main/kotlin/Dependencies.kt [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+    UI("com.tailoredapps.androidutil:util-ui:${Versions.AndroidAppUtil}"),
+    Async("com.tailoredapps.androidutil:util-async:${Versions.AndroidAppUtil}"),
+    Network("com.tailoredapps.androidutil:util-network:${Versions.AndroidAppUtil}"),
+    Optional("com.tailoredapps.androidutil:util-optional:${Versions.AndroidAppUtil}"),
+    Permissions("com.tailoredapps.androidutil:util-permissions:${Versions.AndroidAppUtil}"),
+    ViewState("com.tailoredapps.androidutil:util-viewstate:${Versions.AndroidAppUtil}");
+    
+    override fun toString() = dep
```
Comment (FlowMo7) [inline buildSrc/src/main/kotlin/Dependencies.kt:57]: Really nice solution! :+1:

**Review by FlowMo7 — DISMISSED** [review]
Nice!

**Comment by AlexGrafl** [comment]
TODO: revert all but test

File: buildSrc/src/main/kotlin/Dependencies.kt (hunk 1 above)
Comment (AlexGrafl) [reply buildSrc/src/main/kotlin/Dependencies.kt:49]: ok

**Pull request merged**